File: Kotlin/src/main/java/no/agitec/fagaften/mars/kotlin/section09/calljava1/javacode/Car.java
package no.agitec.fagaften.mars.kotlin.section09.calljava1.javacode;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class Car {

    private String color;
    private String model;
    private String modelA;
    private String modelA2;
    private int year;

    public Car(String color, String model, int year) {
        this.color = color;
        this.model = model;
        this.year = year;
    }

    public String getColor() {
        return color;
    }

    public void setColor(@NotNull String color) {
        this.color = color;
    }

    public  String getModel() {
        return model;
    }

    public @Nullable
    String getModelA() {
        return modelA;
    }

    public @NotNull String getModelA2() {
        return modelA2;
    }

    public void setModel(/*@NotNull*/ String model) {
        this.model = model;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    @Override
    public String toString() {
        return "Car{" +
                "color='" + color + '\'' +
                ", model='" + model + '\'' +
                ", year=" + year +
                '}';
    }
}

File: Kotlin/src/main/java/no/agitec/fagaften/mars/kotlin/section09/kotlincode1/Car.kt
@file:JvmName("StaticCar")

package no.agitec.fagaften.mars.kotlin.section09.kotlincode1

fun topLevel() = println("I'm in the Car file!")

fun main(args: Array<String>) {
    "Print this".print()
    //Print this
}

class Car(val color: String, @JvmField val model: String, val year: Int) {

//    var color: String = color
//        private set(value) {
//            field = "always green"
//        }

}

fun String.print() {
    println(this)
}